File: SpringBoot-SecKill-Scala/src/main/scala/cn/edu/jxnu/seckill/dao/SeckillUserDao.scala
package cn.edu.jxnu.seckill.dao

import scala.language.implicitConversions
import org.apache.ibatis.annotations.Select
import org.apache.ibatis.annotations.Param
import cn.edu.jxnu.seckill.domain.SeckillUser
import org.apache.ibatis.annotations.Update

/**
 * 秒杀用户dao
 *
 * @author 梦境迷离.
 * @time 2018年5月19日
 * @version v1.0
 */
trait SeckillUserDao {

    /**
     * 根据id查询秒杀用户
     */
    @Select(Array("select * from seckill_user where id = #{id}"))
    def getById(@Param("id") id: Long): SeckillUser

    /**
     * 根据id和密码跟新秒杀用户
     */
    @Update(Array("update seckill_user set password = #{password} where id = #{id}"))
    def update(toBeUpdate: SeckillUser)
}

File: seckill/src/main/java/cn/edu/jxnu/seckill/dao/SeckillUserDao.java
package cn.edu.jxnu.seckill.dao;

import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;
import org.apache.ibatis.annotations.Update;

import cn.edu.jxnu.seckill.domain.SeckillUser;

@Mapper
public interface SeckillUserDao {

    @Select("select * from seckill_user where id = #{id}")
    SeckillUser getByid(@Param("id") long id);

    @Update("update seckill_user set password = #{password} where id = #{id}")
    void update(SeckillUser toBeUpdate);
}